apidef/oas/default.go

```diff
@@ -0,0 +1,61 @@
+package oas
+
+import (
+	"fmt"
+	neturl "net/url"
+)
+
+type TykExtensionConfigParams struct {
+	UpstreamURL string
+	ListenPath  string
+}
+
+func (s *OAS) BuildDefaultTykExtension(overRideValues TykExtensionConfigParams) error {
+	xTykAPIGateway := s.GetTykExtension()
+
+	if xTykAPIGateway == nil {
+		xTykAPIGateway = &XTykAPIGateway{}
+		s.SetTykExtension(xTykAPIGateway)
+	}
+
+	if xTykAPIGateway.Info.Name == "" {
+		xTykAPIGateway.Info.Name = s.Info.Title
+	}
+
+	xTykAPIGateway.Info.State.Active = true
+	xTykAPIGateway.Info.State.Internal = false
+
+	if overRideValues.ListenPath != "" {
+		xTykAPIGateway.Server.ListenPath.Value = overRideValues.ListenPath
+	} else if xTykAPIGateway.Server.ListenPath.Value == "" {
```

Comment:
Trying to respect already existing listenPath value in xTyk extension. For sure at most priority goes to the param `listenPath`, in case if it's not present, isn't better to default listenPath to `/` when the listenPath is not configured in xTyk extension?